### ISSUE TITLE
Parse error in tuple access  `foo.0. bar`

### ISSUE THREAD

**Issue by ogoffart** [rust-lang/rust-analyzer#10492]
This is supported rust syntax since rust 1.46, but rust-analyzer fail to parse it in some cases:

The following code compiles fine, but rust-analizer incorrectly reports an error:
```rust
fn main() {
    let foo = (String::new(),);
    foo.0.as_str(); // OK
    foo.0. as_str(); // ERROR in rust-analyzer! (Syntax Error: expected SEMICOLON);
}
```

**Comment by ogoffart** [comment]
Context: you might argue that this is incorrect style, but in my case, rust-analyzer was reporting error in auto-generated code using quote. It was generated with something like the following exampe
```
fn main() {
   let xx = quote::quote!(as_str());
   println!("{}", quote::quote!(foo.0.#xx));
}
```
which prints ` foo . 0. as_str ()`
